src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java

```diff
@@ -0,0 +1,3122 @@
+			if (retryOnError) {
+				retrieveDevices(false);
+			}
+			logger.error(String.format("Aggregated Device Data Retrieval-Error: %s with cause: %s", e.getMessage(), e.getCause().getMessage()), e);
+		}
+	}
+
+	//--------------------------------------------------------------------------------------------------------------------------------
+	//endregion
+
+	//region perform aggregator control: location, room, create room, scene
+	//--------------------------------------------------------------------------------------------------------------------------------
+
+	/**
+	 * This method is used for calling control location properties:
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 * @param controllableProperty name of controllable property
+	 * @param value value of controllable property
+	 * @throws ResourceNotReachableException when fail to control
+	 */
+	private void locationControl(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, String controllableProperty, String value) {
+		try {
+			int locationIndex = Integer.parseInt(controllableProperty.substring(LocationManagementMetric.LOCATION.getName().length()));
+			for (int i = 0; i < cachedLocations.size(); i++) {
+				if (i == locationIndex) {
+					continue;
+				}
+				if (cachedLocations.get(i).getName().equalsIgnoreCase(value)) {
+					throw new ResourceNotReachableException(String.format("The location name %s already exists, Please chose the different location name", value));
+				}
+			}
+			String request = SmartThingsURL.LOCATIONS
+					.concat(SmartThingsConstant.SLASH)
+					.concat(cachedLocations.get(locationIndex).getLocationId());
+
+			HttpHeaders headers = new HttpHeaders();
+
+			Location location = new Location();
+			location.setName(value);
+			String requestBody = location.contributeRequestBody();
+
+			ResponseEntity<?> response = doRequest(request, HttpMethod.PUT, headers, requestBody, String.class);
+
+			handleRateLimitExceed(response);
+
+			Optional<?> responseBody = Optional.ofNullable(response)
+					.map(HttpEntity::getBody);
+			if (response.getStatusCode().is2xxSuccessful() && responseBody.isPresent()) {
+				location = objectMapper.readValue(responseBody.get().toString(), Location.class);
+				cachedLocations.set(locationIndex, location);
+			} else {
+				throw new ResourceNotReachableException(String.format("Changing %s name fail, please try again later", controllableProperty));
+			}
+
+			addAdvanceControlProperties(advancedControllableProperties,
+					createText(stats, AggregatorGroupControllingMetric.LOCATION_MANAGEMENT.getName() + LocationManagementMetric.LOCATION.getName() + locationIndex, value));
+			isEmergencyDelivery = true;
+		} catch (Exception e) {
+			throw new ResourceNotReachableException(String.format("Error while controlling location %s: %s", controllableProperty, e.getMessage()), e);
```

Comment:
This error could also use some specifics like locationName for instance. Would make it a bit more clear at glance, plus we can't be sure that error message would include that

Reply:
I fixed it